realm/realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -254,6 +259,9 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeConvertColumnToNullabl
                 case type_Table:
                     // checked previously
                     break;
+                case type_OldDateTime:
+                    // not used
```

Comment:
It should not be possible. All old Realm files are converted, and new Realm will only use `Timestamp`.